tidb-performance-tuning-config.md

```diff
@@ -0,0 +1,533 @@
+
+### TiKV configurations
+
+Add the following configuration items to the TiKV configuration file:
+
+```toml
+[server]
+concurrent-send-snap-limit = 64
+concurrent-recv-snap-limit = 64
+snap-io-max-bytes-per-sec = "400MB"
+
+[pessimistic-txn]
+in-memory-peer-size-limit = "32MiB"
+in-memory-instance-size-limit = "512MiB"
+
+[rocksdb.titan]
+enabled = true
+[rocksdb.defaultcf.titan]
+min-blob-size = "1KB"
+blob-file-compression = "zstd"
+
+[storage.flow-control]
+l0-files-threshold = 60
+```
+
+| Configuration item | Description | Note |
+| ---------| ---- | ----|
+| [`concurrent-send-snap-limit`](/tikv-configuration-file.md#concurrent-send-snap-limit), [`concurrent-recv-snap-limit`](/tikv-configuration-file.md#concurrent-recv-snap-limit), and [`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec) | Set limits for concurrent snapshot transfer and I/O bandwidth during TiKV scaling operations. Higher limits reduce scaling time by allowing faster data migration. | Adjusting these limits affects the trade-off between scaling speed and online transaction performance. |
+| [`in-memory-peer-size-limit`](/tikv-configuration-file.md#in-memory-peer-size-limit-new-in-v840) and [`in-memory-instance-size-limit`](/tikv-configuration-file.md#in-memory-instance-size-limit-new-in-v840) | Control the memory allocation for pessimistic lock caching at the Region and TiKV instance levels. Storing locks in memory reduces disk I/O and improves transaction performance. | Monitor memory usage carefully. Higher limits improve performance but increase memory consumption. |
+| [`rocksdb.titan`](/tikv-configuration-file.md#rocksdbtitan), [`rocksdb.defaultcf.titan`](/tikv-configuration-file.md#rocksdbdefaultcftitan), [`min-blob-size`](/tikv-configuration-file.md#min-blob-size), and [`blob-file-compression`](/tikv-configuration-file.md#blob-file-compression) | Enable the Titan storage engine to reduce write amplification and alleviate disk I/O bottlenecks. Particularly useful when RocksDB compaction cannot keep up with write workloads, resulting in accumulated pending compaction bytes. | Enable it when write amplification is the primary bottleneck. Trade-offs include: 1. Potential performance impact on primary key range scans. 2. Increased space amplification (up to 2x in the worst case). 3. Additional memory usage for blob cache. |
+| [`storage.flow-control.l0-files-threshold`](/tikv-configuration-file.md#l0-files-threshold) | Control when write flow control is triggered based on the number of kvDB L0 files. Increasing the threshold reduces write stalls during high write workloads. | Higher thresholds might lead to more aggressive compactions when many L0 files exist. |
+
+### TiFlash configurations
+
+Add the following configuration items to the TiFlash configuration file:
+
+```toml
+[raftstore-proxy.server]
```

Comment:
Cannot find the `snap-io-max-bytes-per-sec` parameter in the [TiFlash confi file](https://docs.pingcap.com/tidb/v8.4/tiflash-configuration). It's only in the [TiKV config file](https://docs.pingcap.com/tidb/v8.4/tikv-configuration-file#snap-io-max-bytes-per-sec).
Do you mean `max_write_bytes_per_sec`?